### PR TITLE
BKNDLSS-21176 support for DataQueryBuilder in FindById

### PR DESCRIPTION
**Pull request by holailusoria**

### CODE REVIEW

File: Projects/TestProject/TestInitialization.cs [hunk 1]
```diff
@@ -11,12 +11,12 @@ namespace TestProject
...
 #if DEV_TEST
-      private const String APP_API_KEY = "F8F82BF0-414F-36CB-FFE1-303FA538ED00";
+      public const String APP_API_KEY = "F8F82BF0-414F-36CB-FFE1-303FA538ED00";
       private const String DOTNET_API_KEY = "AB320716-358B-4BBF-AF9D-4F4B98F03363";
       private const String BKNDLSS_URL = "http://apitest.backendless.com";
 #else
-      public const String APP_API_KEY = "B5D20616-5565-2674-FF73-C5CAC72BD200";
-      private const String DOTNET_API_KEY = "18BF3443-B8A8-48E1-90ED-2783F9AF2D40";
+      public const String APP_API_KEY = "CF7398E8-7F3D-64F9-FF3B-FB3CE8893200";
+      private const String DOTNET_API_KEY = "9808682C-32C2-EC87-FFF5-CAB709BE2400";
```
Comment (markpiller) [inline Projects/TestProject/TestInitialization.cs:19]: remove api key from the code, add it here whenever you run tests locally

Comment (markpiller) [reply Projects/TestProject/TestInitialization.cs:19]: same for the apitest deployment

File: Backendless/Data/IDataStore.cs [hunk 2]
```diff
@@ -123,6 +127,8 @@ public interface IDataStore<T>
     Task<T> FindByIdAsync( T entity, IList<string> relations, int? relationsDepth );
   #endif
     void FindById( string id, AsyncCallback<T> responder );
+    void FindById( string id, DataQueryBuilder queryBuilder, AsyncCallback<T> responder );
+    void FindById( T entity, DataQueryBuilder queryBuilder, AsyncCallback<T> responder );
```
Comment (markpiller) [inline Backendless/Data/IDataStore.cs:131]: does this method override cause a conflict with the method which accepts string?

File: Backendless/Data/DictionaryDrivenDataStore.cs [hunk 3]
```diff
@@ -357,6 +357,26 @@ public void Find( DataQueryBuilder dataQueryBuilder, AsyncCallback<IList<Diction
       return FindById( id, 0 );
     }
 
+    public Dictionary<string, object> FindById( string id, DataQueryBuilder queryBuilder )
```
Comment (markpiller) [inline Backendless/Data/DictionaryDrivenDataStore.cs:360]: here and everywhere else let's change `string` to `String` for uniformity

File: Backendless/Data/IDataStore.cs (hunk 2 above)
Comment (holailusoria) [reply Backendless/Data/IDataStore.cs:131]: @markpiller no

**Pull request merged**